### ISSUE TITLE
None is set to 1 when import TF model

### ISSUE THREAD

**Issue by SidneyLann** [deeplearning4j/deeplearning4j#8561]
#### Issue Description
Please describe our issue, along with:
- expected behavior
Dimension None should be set to -1

- encountered behavior
Dimension None is set to 1

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version
beta6
* Platform information (OS, etc)
* CUDA version, if used
* NVIDIA driver version, if in use

#### Additional Information
--- Variables ---
- Name -                                                                             - Array Shape -     - Variable Type -   - Data Type-        - Output Of Function -                                                                       - Inputs To Functions -
Placeholder                                                                          [1, 128]           PLACEHOLDER         INT                 <none>                                                                                       [bert/embeddings/ExpandDims]
Placeholder_1                                                                        [1, 128]           PLACEHOLDER         INT                 <none>                                                                                       [bert/encoder/Reshape]
Placeholder_2                                                                        [1, 128]           PLACEHOLDER         INT                 <none>                                                                                       [bert/embeddings/Reshape_1]
bert/embeddings/ExpandDims/dim                                                       [1]                 CONSTANT            INT                 <none>                                                                                       [bert/embeddings/ExpandDims]
bert/embeddings/word_embeddings                                                      [21128, 768]        CONSTANT            FLOAT               <none>                                                                                       [bert/embeddings/word_embeddings/read]
bert/embeddings/embedding_lookup/axis                                                []                  CONSTANT            INT                 <none>                                                                                       [bert/embeddings/embedding_lookup]
bert/embeddings/Reshape/shape                                                        [3]                 CONSTANT            INT                 <none>                                                                                       [bert/embeddings/Reshape]
bert/embeddings/token_type_embeddings                                                [2, 768]            CONSTANT            FLOAT               <none>                                                                                       [bert/embeddings/token_type_embeddings/read]
bert/embeddings/Reshape_1/shape                                                      [1]                 CONSTANT            INT                 <none>                                                                                       [bert/embeddings/Reshape_1]

**Comment by SidneyLann** [comment]
1.x model not saved model format

**Comment by SidneyLann** [comment]
Maybe py translate None to 1 when convert checkpoint to pb

**Comment by AlexDBlack** [comment]
Where is this model from?
If it's the official BERT repo (here: https://github.com/google-research/bert ), they unfortunately made (a very bad IMO) design decision to have the model use a fixed minibatch size only.
There's nothing in the transformer model itself that requires a fixed minibatch size; it's something specific to their implementation. When working with BERT, I have looked at changing that in SameDiff, but it's not just the placeholder shape, it's many places all throughout the model.

**Comment by SidneyLann** [comment]
yes, I download from  https://github.com/google-research/bert.

**Comment by SidneyLann** [comment]
Is below model trained fine to use? https://dl4jdata.blob.core.windows.net/testresources/bert_mrpc_frozen_v1.zip?

And where is it from? It has the different placeholder names between offical one, it has also fixed mini batch size!

**Comment by AlexDBlack** [comment]
Right, so this is what I expected - the model has a hardcoded minibatch size at the TensorFlow level.
Unfortunately other than trying to "fix" the model manually, there's not much we can do here. It's not an import problem, the SameDiff model is imported correctly, and is the same as the TF model.

>Is below model trained fine to use? https://dl4jdata.blob.core.windows.net/testresources/bert_mrpc_frozen_v1.zip?

That's converted from the same repo that you used. IIRC it has fixed minibatch size of 4 instead of 1.

**Comment by AlexDBlack** [comment]
As for that model - it was produced using the instructions here: https://github.com/KonduitAI/dl4j-dev-tools/tree/master/import-tests/model_zoo/bert
